client/src/components/App.jsx

```diff
@@ -1,23 +1,60 @@
 import React from 'react';
 import ReactDom from 'react-dom';
 import ListingDetail from './ListingDetail.jsx';
+import axios from 'axios';
+const Promise = require('bluebird');
+
+const API_URL = 'http://localhost:3005';
 
 class App extends React.Component {
-  constructor() {
-    super();
+  constructor(props) {
+    super(props);
     this.state = {
       listingData: {},
-      features: []
+      featuresData: []
     };
   }
+
+  fetchListingData(id) {
+    const self = this;
```

Comment:
I get what your reasoning is, but I feel like Magee would not approve of variable naming this to self, because it's an unusual pattern and makes your code less readable.